src/main/java/iudx/catalogue/server/authenticator/JwtAuthenticationServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+  }
+
+  Future<Boolean> isValidId(JwtData jwtData, String id) {
+    Promise<Boolean> promise = Promise.promise();
+
+    // id is the (substring [1] of Iid with delimiter as ':' )'s substring [1] with delimiter as '/'
+    String jwtId = (jwtData.getIid().split(":")[1]).split("/")[1];
+
+    if (id.equalsIgnoreCase(jwtId)) {
+      promise.complete(true);
+    } else {
+      LOGGER.error("Incorrect id value in jwt");
+      promise.fail("Incorrect id value in jwt");
+    }
+    return promise.future();
+  }
+
+  Future<Boolean> isValidEndpoint(String endPoint) {
+    Promise<Boolean> promise = Promise.promise();
+
+    if(endPoint.equals(ITEM_ENDPOINT) || endPoint.equals(INSTANCE_ENDPOINT)) {
+      promise.complete(true);
+    } else {
+      LOGGER.error("Incorrect endpoint in jwt");
+      promise.fail("Incorrect endpoint in jwt");
+    }
+    return promise.future();
+  }
+
+  public Future<JsonObject> validateAccess(JwtData jwtData, JsonObject authInfo) {
+    LOGGER.trace("validateAccess() started");
+    Promise<JsonObject> promise = Promise.promise();
+
+    Method method = Method.valueOf(authInfo.getString(METHOD));
+    Api api = Api.fromEndpoint(authInfo.getString(API_ENDPOINT));
+
+    AuthorizationRequest authRequest = new AuthorizationRequest(method,api);
+
+    AuthorizationStratergy authStrategy = AuthorizationContextFactory.create(jwtData.getRole());
+    LOGGER.info("strategy: " + authStrategy.getClass().getSimpleName());
+
+    JwtAuthorization jwtAuthStrategy = new JwtAuthorization(authStrategy);
+    LOGGER.info("endpoint: "+authInfo.getString(API_ENDPOINT));
+
+    if(jwtAuthStrategy.isAuthorized(authRequest, jwtData)) {
+      LOGGER.info("User access is allowed");
+      JsonObject response = new JsonObject().put(JSON_USERID, jwtData.getSub());
+      promise.complete(response);
+    } else {
+      LOGGER.info("user access denied");
+      JsonObject result = new JsonObject().put("401","no access provided to endpoint");
+      promise.fail(result.toString());
+    }
+    return promise.future();
+  }
+
+
+  @Override
+  public AuthenticationService tokenInterospect(JsonObject request, JsonObject authenticationInfo,
+                                                Handler<AsyncResult<JsonObject>> handler) {
+    String endPoint = authenticationInfo.getString(API_ENDPOINT);
+    String id = authenticationInfo.getString(ID);
+    String token = authenticationInfo.getString(TOKEN);
+
+    Future<JwtData> jwtDecodeFuture = decodeJwt(token);
+
+    ResultContainer result = new ResultContainer();
+
+    jwtDecodeFuture.compose(decodeHandler -> {
+      result.jwtData =  decodeHandler;
+      return isValidAudienceValue(result.jwtData);
+    }).compose(audienceHandler -> {
+      return isValidId(result.jwtData, id);
+    }).compose(validIdHandler -> {
+      return isValidEndpoint(endPoint);
+    }).compose(validEndpointHandler -> {
+      return validateAccess(result.jwtData, authenticationInfo);
```

Comment:
What does authenticationInfo contain?
Is this where the final match between the person who has issued the token and the "provider" field (in case of Post) or "id" field (in case of DELETE) is made?